tests/tests_fabric/strategies/test_fsdp_integration.py

```diff
@@ -305,7 +305,7 @@ def test_setup_module_move_to_device(fabric_module_mock, move_to_device):
         assert isinstance(next(fabric_model.parameters()), FlatParameter)
 
     # The _DeviceDtypeModuleMixin currently can't represent the device in a meaningful way for sharded models
-    assert fabric_model.device == torch.device("cpu")
+    assert fabric_model.device.type == "cuda"
```

Comment:
nit: you could put the same assertion as below

Reply:
I didn't do this because for a real model, the index would depend on the sharding from layers to devices.

Since the test only uses 1 Linear, it's not a concern.

```suggestion
    assert fabric_model.device == torch.device("cuda", fabric.local_rank)
```

Please commit the suggestion if you still want it to be equal.

The comment in the line above might need tweaking too, because the assertion doesn't convey what it says anymore

Reply:
Even in the case of sharded models the device would be well defined because a process / GPU would only ever handle one device, i.e., the local rank across the entire module and submodules. So the comment was always a bit wrong. 

The issue would be if submodules were on different devices.